apply-functional-programming-to-convert-strings-to-url-slugs.js

```diff
@@ -0,0 +1,25 @@
+// Only change code below this line
+function urlSlug(title) {
```

Comment:
```suggestion
// Only change code below this line
function urlSlug(title) {
  // Separar palabras, quitando espacios repetidos
  var urlArray = title.toLowerCase().split(/\s+/).reduce((cleanArray, word) => {
    // Si hay caracteres no alfanuméricos
    if (word.match(/\W/)) {
      // Limpiar caracteres no alfanuméricos
      word = word.split("").filter(item => item.match(/\w/)).join(""); 
    }
    cleanArray.push(word);
    
    return cleanArray;
  }, []).filter(item => item != ""); //quitar huecos

  return urlArray.join("-");
}
// Only change code above this line
```

Comments should go above the code they are commenting, it's quite illegible if they are at the end of the line.

Cleaning your code I'm left with this, but it still seems quite messy. Could you just use the reduce to just iterate and clean through the whole string withouth first splitting it? I think that might be cleaner.

Reply:
You are probably right. I'm still having something of a hard time with reduce() and tend to favor other methods instead.